towntest_chest/init.lua

```diff
@@ -61,14 +61,62 @@ towntest_chest.load = function(filename)
 	return towntest_chest.get_string(building)
 end
 
+
+local function mapname(name)
+	local node = minetest.registered_items[name]
+
+	if not node then
+		minetest.log("info", "unknown node in building: "..name)
+		return nil
+	else
+
+-- known node. Check for price or if it is free
+		if (node.groups.not_in_creative_inventory and not (node.groups.not_in_creative_inventory == 0)) or
+		   (not node.description or node.description == "") then
+			if node.drop then
+				return mapname(node.drop) --use the drop as payment
+			else --something not supported, but known
+				return "free" -- will be build for free. they are something like doors:hidden or second part of coffee lrfurn:coffeetable_back
+			end
+		else
+			return node.name
+		end
+	end
+end
+
 -- get_table - convert building table to string
 -- building_string - string containing pos and nodes to build
 -- return - table containing pos and nodes to build
 towntest_chest.get_table = function(building_string)
 	local building = {}
-	for x, y, z, name, param1, param2 in building_string:gmatch("([+-]?%d+)%s+([+-]?%d+)%s+([+-]?%d+)%s+([^%s]+)%s+(%d+)%s+(%d+)[^\r\n]*[\r\n]*") do
-		if tonumber(x)~=0 or tonumber(y)~=0 or tonumber(z)~=0 then
-			table.insert(building, {x=x,y=y,z=z,name=name,param1=param1,param2=param2})
+        local wefile = {}
+	local idx, def
+	local retpos = string.find(string.sub(building_string,0,10), "return")
+	if retpos then
+		local exe, err, ok
```

Comment:
Well, then we'll support the latest version only. I think Minetest development got stuck by wanting to stay backward compatible at all costs. My server tracks git master, so... I'm for the simplest and clearer solution with `minetest.deserialize`.